### PR TITLE
util/strutil: do not panic on wrong duration unit.

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

File: util/strutil/strconv.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func StringToDuration(durationStr string) (duration time.Duration, err error) {
 	case "s":
 		duration *= 1
 	default:
-		panic("Invalid time unit in duration string.")
+		return 0, fmt.Errorf("Invalid time unit %q in duration string.", unit)
```
Comment (brian-brazil) [inline util/strutil/strconv.go]: This is inconsistent with how it's done earlier in the function

Comment (fabxc) [inline util/strutil/strconv.go]: Fixed.

File: util/strutil/strconv.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func StringToDuration(durationStr string) (duration time.Duration, err error) {
 	case "s":
 		duration *= 1
 	default:
-		panic("Invalid time unit in duration string.")
+		return 0, fmt.Errorf("Invalid time unit in duration string: %q", unit)
```
Comment (beorn7) [inline util/strutil/strconv.go]: Error strings should not be capitalized (correctly done in line 61).

https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Error_Strings

**Comment by beorn7** [comment]
:+1:

**Pull request merged**